src/views/QueryEditor/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useEffect, useState } from 'react';
+import { EditorMode } from '../../../../types/types';
+import { Button, InlineField, InlineFieldRow, InlineLabel, Select } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+
+const fetchData = async (url: string, setter: (value: any) => void) => {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
+    }
+
+    const data = await response.json();
+    setter(data.data.map((item: any) => ({ label: item.name, value: item.name })));
+  } catch (error) {
+    console.error(`Error fetching data from ${url}:`, error);
+  }
+};
+
+export const QueryBuilder = ({ query, onRunQuery, onChange, datasource }: any) => {
+  const [databases, setDatabases] = useState([]);
+  const [tables, setTables] = useState([]);
+  const [dateTimeColumns, setDateTimeColumns] = useState([]);
+  const [timestampColumns, setTimestampColumns] = useState([]);
+  const [selectedDatabase, setSelectedDatabase] = useState<SelectableValue<string>>();
+  const [selectedTable, setSelectedTable] = useState<string>();
+  const [selectedColumnTimestampType, setSelectedColumnTimestampType] = useState([]);
+
+  useEffect(() => {
+    fetchData('http://localhost:8123/?query=SHOW%20DATABASES%20FORMAT%20JSON', setDatabases);
+  }, []);
+
+  useEffect(() => {
+    if (selectedDatabase) {
+      fetchData(`http://localhost:8123/?query=SHOW%20TABLES%20FROM%20${selectedDatabase}%20FORMAT%20JSON`, setTables);
```

Comment:
Please remove hard-code localhost here
it shall be simple SQL query instead of url encoded values